clang/lib/CodeGen/TargetBuiltins/WebAssembly.cpp

```diff
@@ -246,35 +246,26 @@ Value *CodeGenFunction::EmitWebAssemblyBuiltinExpr(unsigned BuiltinID,
     llvm::FunctionType *LLVMFuncTy =
         cast<llvm::FunctionType>(ConvertType(QualType(FuncTy, 0)));
 
+    bool VarArg = LLVMFuncTy->isVarArg();
     unsigned NParams = LLVMFuncTy->getNumParams();
     std::vector<Value *> Args;
-    Args.reserve(NParams + 3);
+    Args.reserve(NParams + 3 + VarArg);
```

Comment:
I see you have a test for calling the intrinsic below, but: Vararg functions basically have a pointer to the vararg buffer as their last argument, so I'm wondering whether we need to take any more special cases to distinguish between that vs a function that just has a pointer in the last position. Those would be compatible in wasm, but not in C. But I guess the builtin/intrinsic is specifically for testing wasm compatibility vs C compatibility, right?

Reply:
Yeah it's just testing for wasm compatibility. I don't think it's possible to test for C compatibility since the runtime doesn't know about that.